pkgs/tools/security/rustscan/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchFromGitHub
+, rustPlatform
+, makeWrapper
+, nmap
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rustscan";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "RustScan";
+    repo = pname;
+    rev = "${version}";
+    sha256 = "0rkqsh4i58cf18ad97yr4f68s5jg6z0ybz4bw8607lz7cjkfvjay";
+  };
+
+  cargoSha256 = "0mj214f2md7kjknmcayc5dcfmlk2b8mqkn7kxzdis8qv9a5xcbk8";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  fixupPhase = ''
+    wrapProgram $out/bin/rustscan --prefix PATH ":" ${nmap}/bin ;
+  '';
```

Comment:
From an admittedly very quick look, it seems that `nmap` is only hardcoded in the source once. You could patch this up and then you do not need a wrapper (which is always nice):

```
  postPatch = ''
    substituteInPlace src/main.rs \
      --replace 'Command::new("nmap")' 'Command::new("$out/bin/nmap")'
  '';
```

Untested, but just to give an idea of how you could do this.

Reply:
Took me a few tries but I got it working. Thanks!